File: 1829-maximum-units-on-a-truck/maximum-units-on-a-truck.cpp
class Solution {
public:
    static bool cmp(vector<int>&a , vector<int>&b){
        if(a[1] == b[1]){
            return a[0] > b[0];
        }
        return a[1] > b[1];
    }
    int maximumUnits(vector<vector<int>>& arr, int k) {
        int n = arr.size();
        sort(arr.begin() , arr.end() , cmp);

        int sum = 0;

        int i = 0;
        while(i < n){
            if(k >= arr[i][0]){
                sum += arr[i][0] * arr[i][1];
                k = k - arr[i][0];

            }
            else{
                sum += k*arr[i][1];
                k = 0;
            }

            i++;
            if(k == 0)break;
        }
        return sum;

        
    }
};

File: 1829-maximum-units-on-a-truck/maximum-units-on-a-truck.java
class Solution {
    public int maximumUnits(int[][] arr, int k) {
        int n = arr.length;
        Arrays.sort(arr , (a , b) -> {
            if(a[1] == b[1]){
                return b[0] - a[0];
            }
            return b[1] - a[1];
        });
        int cnt = 0;

        for(int i = 0 ; i < n ; i++){
            if(arr[i][0] <= k){
                cnt += arr[i][0] * arr[i][1];
                k -= arr[i][0];                
            }
            else{
                cnt += k * arr[i][1];
                k = 0;
            }
            if(k == 0)break;
        }
        return cnt;


        
    }
}